### PR TITLE
bug: Main thread dispatcher

### PR DESCRIPTION
**Pull request by asd1989dsa**
## Purpose of this PR
Fix to prevent freezing

## Testing status
* No tests have been added.

User can measure the method execution speed and set the maximum available execution time

### CODE REVIEW

**Comment by asd1989dsa** [comment]
User can measure the method execution speed and set the maximum available execution time

**Comment by PavelKlymentenkoSA** [comment]
image

File: Runtime/Async/MainThreadDispatcher/MainThreadDispatcherBase.cs [hunk 1]
```diff
@@ -1,20 +1,30 @@
...
         protected void Update()
         {
-            if(s_ExecutionQueue.TryDequeue(out var action))
+            while (s_ExecutionQueue.TryDequeue(out var action) && sw.ElapsedMilliseconds < maxMillisecondsExecution)
+            {
+                sw.Start();
                 action.Invoke();
+                sw.Restart();
+                Debug.Log(sw.ElapsedMilliseconds);
+            }
```
[the remark this answers is not in the document]
Comment (PavelKlym) [reply Runtime/Async/MainThreadDispatcher/MainThreadDispatcherBase.cs:27]: I think it's okay.

[the remark this answers is not in the document]
Comment (PavelKlym) [reply Runtime/Async/MainThreadDispatcher/MainThreadDispatcherBase.cs:27]: Removed useless log.

**Pull request closed without merging**